lib/active_record/cursor_paginator.rb

```diff
@@ -264,6 +263,29 @@ def build_filter_query(sorted_relation, op, current_field, prev_fields)
         relation.where("#{col} #{op} ?", val)
       end
 
+      # parse aliases from select values
+      #
+      # @return [Hash]
+      def parse_aliases
+        aliases = {}
+        @relation.select_values.each do |select_value|
+          next unless select_value.is_a? String
+
+          select_value.split(',').each do |expr|
+            expr.strip!
+            # "value [AS|as] alias" という形式に対応する
+            # value: spaceがあってもOK. 最小マッチングのため '?' をつける
+            # 'as' に後方参照させないため ?: をつける
+            match = expr.match(/^(.+?)\s+(?:[Aa][Ss]\s+)?(\S+)$/)
```

Comment:
```suggestion
            match = expr.match(/^(?<value>.+?)\s+(?:[Aa][Ss]\s+)?(?<alias>\S+)$/)
```
名前付きでキャプチャしておくと想定する挙動と参照するコードがわかりやすくなるかと思います。

```ruby
key = trim_quote(match[:alias])
aliases[key] = match[:value]
```